### PR TITLE
ref(ui) Extract repository add/delete/cancel to action creators

### PR DESCRIPTION
**Pull request by markstory**
Move repository related actions to action creators & promises. I've co-located the indicator operations with the API requests so that the react components can be more focused on their state management and have fewer concerns intertwined.

### CODE REVIEW

File: src/sentry/static/sentry/app/actionCreators/integrations.jsx [hunk 1]
```diff
@@ -49,3 +50,100 @@ export function addIntegrationToProject(orgId, projectId, integration) {
...
+ * @param {Object} client ApiClient
+ * @param {String} orgId Organization Slug
+ * @param {Number} repositoryId Repository ID
+ */
+export function deleteRepository(client, orgId, repositoryId) {
+  addLoadingMessage();
+  let promise = client.requestPromise(`/organizations/${orgId}/repos/${repositoryId}/`, {
+    method: 'DELETE',
+  });
+  promise.then(
```
Comment (markstory) [inline src/sentry/static/sentry/app/actionCreators/integrations.jsx]: I intentionally didn't return the chain result here as I didn't want to have to throw an error from the catch callback.

File: src/sentry/static/sentry/app/actionCreators/integrations.jsx [hunk 2]
```diff
@@ -49,3 +50,100 @@ export function addIntegrationToProject(orgId, projectId, integration) {
...
+    () => clearIndicators(),
+    () => addErrorMessage(t('Unable to delete repository.'))
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/actionCreators/integrations.jsx]: Note that `addErrorMessage` will clear by default. Or did you want to explicitly clear for some other reason?

Comment (markstory) [reply src/sentry/static/sentry/app/actionCreators/integrations.jsx]: I wanted to clear in the success case, so the 'loading' message doesn't stick around forever.

File: src/sentry/static/sentry/app/actionCreators/integrations.jsx [hunk 3]
```diff
@@ -49,3 +50,100 @@ export function addIntegrationToProject(orgId, projectId, integration) {
     }
   );
 }
+
+/**
+ * Delete a respository
+ *
+ * @param {Object} client ApiClient
+ * @param {String} orgId Organization Slug
+ * @param {Number} repositoryId Repository ID
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/static/sentry/app/actionCreators/integrations.jsx]: I'm wrong here. I'll fix it.

File: src/sentry/static/sentry/app/actionCreators/integrations.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/actionCreators/integrations.jsx]: Might be good to leave this as a code comment.

**Review by billyvg — APPROVED** [review]
lgtm

**Pull request merged**